### DISCUSSION TITLE
Permission Error

### DISCUSSION THREAD

**Discussion by Cookiek3** [ntherning/StorytelReaderMods#11] [Q&A]
Hello, i'm getting an permission error while running PY3 how do i fix this?
![image](https://github.com/user-attachments/assets/53b22202-5748-4357-b7eb-449ef80bb8d1)

**Comment by raymon-m** [comment]
May I ask what OS you are on?
I had this same problem, running on Windows 11. I haven't been able to come up with an actual fix yet, but I did manage to find a workaround. 
You can see the script is trying to copy a file from ...\StorytelReaderMods\mods\gen2\files\.backup\.magisk to \StorytelReaderMods\tmp\work\ramdisk.extracted\.backup\.magisk. You can manually do this copy, and then add a few lines of code to ...\StorytelReaderMods\tools\repack.py
Open the repack.py file in a text editor like notepad or preferably one with line numbers like vscode or notepad++, then add the following lines of code on line 61:
`if(".magisk" in in_file.absolute().as_posix()):
            continue`

You may need to do the same for a file that the script also attempts to copy, by adding these lines right after the previous couple of lines and then also copying the .rmlist file manually:
`if(".rmlist" in in_file.absolute().as_posix()):
            continue`

If your situation is anything like mine, new errors will follow after this, but we'll deal with those if they happen.

**Reply by Cookiek3** [reply]
Yes i'm using W11 also i'l try your workaround

**Comment by ntherning** [comment]
Please try the python scripts in a Linux VM. I cannot recall whether I actually tested these scripts on Windows though I see now that I mention in the README what you need on your system to run them under Windows. Could be that I just hoped they would work...

**Reply by raymon-m** [reply]
Ah, I hadn't considered using Linux. I will definitely try that. If I can find some extra time I'll also see if I can make it win-compatible

**Comment by Cookiek3** [comment]
Hi Raymon, with your solution i still get an error. Is there a way that i
try and do this while you watch with me?

**Reply by raymon-m** [reply]
Also check out the reply by ntherning, maintainer and builder of this repo. Possibly the script wasn't tested on windows and you might have a better shot at running it on Linux. I might take a look at the script at a later time and see if I can make it work on my windows installation, but I can't guarantee when that would be.
